frontend/packages/topology/src/components/graph-view/components/nodes/WorkloadNode.tsx

```diff
@@ -53,6 +53,14 @@ const WorkloadPodsNode: React.FC<WorkloadPodsNodeProps> = observer(
     const showPodCount = showPodCountFilter?.value ?? false;
     const { decorators } = element.getGraph().getData();
     const nodeDecorators = getNodeDecorators(element, decorators, cx, cy, radius, decoratorRadius);
+
+    const podSet = React.useMemo(() => {
+      if (donutStatus) {
+        return <PodSet size={size} x={cx} y={cy} data={donutStatus} showPodCount={showPodCount} />;
+      }
+      return null;
+    }, [cx, cy, donutStatus, showPodCount, size]);
```

Comment:
I don't see the value in doing this.
I think what you want to do is `React.memo` on `PodSet`.
Just be careful to ensure that we don't mutate data and expect it to re-render.